src/aws-manager.js

```diff
@@ -374,10 +374,11 @@ class AwsManager {
         }
       }));
 
+      // check security group names
       let launchSpecsRegion = worker.instanceTypes.map(t => launchSpecs[r.region][t.instanceType].launchSpec);
-      let allSGForRegion = _.uniq(_.flatten(launchSpecsRegion.map(spec => spec.SecurityGroups)));
+      let allSGForRegion = _.uniq(_.flatten(launchSpecsRegion.map(spec => spec.SecurityGroups).filter(g => g)));
 
-      let hasAllRequiredSG = await sgExists(this.ec2[r.region], allSGForRegion);
+      let hasAllRequiredSG = allSGForRegion.length === 0 || await sgExists(this.ec2[r.region], allSGForRegion);
```

Comment:
I think it'd be a little more clear here if this was an if statement instead of an or